src/lambda/commands/deploySamApplication.ts

```diff
@@ -319,3 +325,43 @@ function getDefaultWindowFunctions(): WindowFunctions {
         showInformationMessage: vscode.window.showInformationMessage,
     }
 }
+
+function notifyDuringLongDeployment(deployParameters: DeploySamApplicationParameters): NodeJS.Timeout {
+    return setTimeout(async () => {
+        const cfnClient = new DefaultCloudFormationClient(deployParameters.region)
+        const stack = (await cfnClient.describeStacks(deployParameters.destinationStackName)).Stacks
+        if (stack !== undefined) {
+            const helpLink = localize('AWS.samcli.deploy.stuckStackHelpLink', 'View docs')
+            const awsConsole = localize(
+                'AWS.samcli.deploy.goToCfnDashboard',
+                'Go to {0} console',
+                getIdeProperties().company
+            )
+            const viewAwsStatus = localize(
+                'AWS.samcli.deploy.viewAwsStatus',
+                'View {0} status',
+                getIdeProperties().company
+            )
+            const cancelUpdate = localize('AWS.samcli.deploy.cancelUpdate', 'Cancel update')
+            const items = [helpLink, awsConsole, viewAwsStatus]
+            if (stack[0].StackStatus === 'UPDATE_IN_PROGRESS') {
+                items.push(cancelUpdate)
+            }
+            const stackStuckMessage = localize(
+                'AWS.samcli.deploy.stackStuckWarning',
+                'Your deployment seems to be taking a while. If this is unexpected, it may be stuck in progress. You may continue to wait or choose an option below.'
+            )
+            vscode.window.showWarningMessage(stackStuckMessage, ...items).then((selection: string | undefined) => {
+                if (selection === helpLink) {
+                    vscode.env.openExternal(vscode.Uri.parse(stuckCfnDeploymentUrl))
+                } else if (selection === awsConsole) {
+                    vscode.env.openExternal(vscode.Uri.parse(awsConsoleUrl))
```

Comment:
It's 100x more useful if we can link to the stack URL in the console. Looks like it has this format (maybe CloudControl can help):

    https://eu-west-1.console.aws.amazon.com/cloudformation/home?region=eu-west-1#/stacks/stackinfo?stackId=arn%3Aaws%3Acloudformation%3Aeu-west-1%3A541201481031%3Astack%2F...

Reply:
The way this URL is formatted creates problems. There is a second `?` after 'stackInfo' in the query section. I've tried multiple ways to create/alter the URI but it always tries to encode the `?`, creating an incorrect link.